File: scripts/check_today_spectrum.py
#!/usr/bin/env python3
import sys
from pymongo import MongoClient

client = MongoClient("mongodb://ryan:Welcome123!@localhost:27017/?authSource=admin")
db = client["radiacode"]

docs = list(
    db.tracker_samples.find(
        {"sessionId": "2026-06-23", "spectrumData": {"$exists": True, "$ne": []}}
    )
    .sort("timestampMs", -1)
    .limit(5)
)

print(f"Found {len(docs)} spectrum docs for today")
for d in docs:
    spec = d.get("spectrumData", [])
    total = sum(spec[:64]) if spec else 0
    nonzero = sum(1 for x in spec if x != 0) if spec else 0
    maxv = max(spec[:64]) if spec else 0
    print(f"  ts={d['timestampMs']} len={len(spec)} total64={total} nonzero={nonzero} max64={maxv}")
    print(f"  first30={spec[:30]}")

File: scripts/check_today_spectrum.js
db = db.getSiblingDB('radiacode');
var docs = db.tracker_samples.find({
  sessionId: '2026-06-23',
  spectrumData: { $exists: true, $ne: [] }
}).sort({ timestampMs: -1 }).limit(5);

print('Found: ' + docs.count() + ' docs');
docs.forEach(function(d) {
  var s = d.spectrumData || [];
  var total64 = s.slice(0, 64).reduce(function(a, b) { return a + b; }, 0);
  var nonzero = s.filter(function(x) { return x !== 0; }).length;
  print('ts=' + d.timestampMs + ' len=' + s.length + ' total64=' + total64 + ' nonzero=' + nonzero + ' sample=' + JSON.stringify(s.slice(0, 25)));
});
